plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/AppEngineConfigFileSchemaLocationProvider.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.validation;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import java.net.URI;
+import java.util.Collections;
+import java.util.Map;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.wst.xml.core.contentmodel.modelquery.IExternalSchemaLocationProvider;
+
+/**
+ * Associate schemas for App Engine standard environment configuration files. Configured from the
+ * {@code org.eclipse.wst.xml.core.externalSchemaLocations} extension point
+ */
+public class AppEngineConfigFileSchemaLocationProvider implements IExternalSchemaLocationProvider {
+  // Relevant XSDs are in this bundle
+  private static final String SCHEMA_LOCATIONS_PREFIX =
+      "platform:/plugin/com.google.cloud.tools.eclipse.appengine.validation/xsd/";
+
+  @Override
+  public Map<String, String> getExternalSchemaLocation(URI fileURI) {
+    Preconditions.checkNotNull(fileURI);
+    IPath path = Path.fromPortableString(fileURI.getPath());
+    if (path.segmentCount() == 0) {
+      return Collections.emptyMap();
+    }
+    String basename = path.lastSegment();
+    switch (basename) {
+      // appengine-web.xml has a known namespace
+      case "appengine-web.xml":
+        return ImmutableMap.of(NO_NAMESPACE_SCHEMA_LOCATION,
+            SCHEMA_LOCATIONS_PREFIX + "appengine-web.xsd", SCHEMA_LOCATION,
```

Comment:
This reads confusing. Better move `SCHEMA_LOCATION,` to the next line.

Reply:
I agree, but google-java-format does not 👎

Reply:
Ah, this is ridiculous. Try line-wrapping all the arguments starting from `NO_NAMESPACE_SCHEMA_LOCATION`. The formatter may tolerate that.